staging/src/k8s.io/legacy-cloud-providers/gce/gce_loadbalancer_internal.go

```diff
@@ -980,3 +996,30 @@ func forwardingRulesEqual(old, new *compute.ForwardingRule) bool {
 		old.AllowGlobalAccess == new.AllowGlobalAccess &&
 		old.Subnetwork == new.Subnetwork
 }
+
+// patchILBNameAnnotation patches the annotations on the service with the provided ilbName.
+// If ilbName is empty or protocol is nil, the forwarding annotation keys will be deleted from
+// resource's annotation map
+func patchILBNameAnnotation(client clientset.Interface, service *v1.Service, protocol *v1.Protocol, ilbName string) (*v1.Service, error) {
+	newService := service.DeepCopy()
+	if newService.Annotations == nil {
+		newService.Annotations = make(map[string]string)
+	}
+
+	// Delete existing keys before setting new value
+	for _, key := range []string{TCPForwardingRuleKey, UDPForwardingRuleKey} {
+		delete(newService.Annotations, key)
+	}
+
+	// If ilbName is empty or protocol is nil, then keys need to be deleted but no annotation is necessary
+	if ilbName != "" && protocol != nil {
```

Comment:
isn't ilbName empty check sufficient? then you can pass in protocol as v1.Protocol instead of pointer.